src/main/java/com/hubspot/jinjava/interpret/JinjavaInterpreter.java

```diff
@@ -907,6 +924,20 @@ private String getWrappedErrorMessage(
     }
   }
 
+  private long clampOutputSizeSafely(long providedLimit) {
+    long configMaxOutput = config.getMaxOutputSize();
+
+    if (configMaxOutput == 0) {
+      return providedLimit;
+    }
+
+    if (providedLimit <= 0) {
+      return configMaxOutput;
+    }
+
+    return Math.min(providedLimit, configMaxOutput);
+  }
```

Comment:
Looks good, can we just get some tests for this logic?

Reply:
Done, thanks a lot Jack.  To avoid repeating tests or changing signatures I moved this into a small static utils class which I tested on it's own.  Let me know if you'd rather I just make it public in the interpreter or add more tests to the render method as a whole and take it back to being private.

I did like this a bit for simplicity though.